### PR TITLE
Dropped use of idamOwner,super-user and admin-user accounts.

### PR DESCRIPTION
**Pull request by ashwini-mv**
### JIRA link ###
https://tools.hmcts.net/jira/browse/SIDM-7109

### Change description ###
Used BeforeSuite functionality to create user accounts utilising random email format:
TEST_IDAM_XXXXXX_USER_XXXXXX@idam.test
Stored default forename, surname and password in config and re-used for each random account
Delete account post-test-suite using teardownAll functionality

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/functional/search-user-test.ts [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
+});
+
 const incorrectEmailAddresses = new DataTable(['incorrectEmailAddress']);
 incorrectEmailAddresses.add(['email']); // adding records to a table
 incorrectEmailAddresses.add(['email@']);
 incorrectEmailAddresses.add(['email@com']);
 
 Data(incorrectEmailAddresses).Scenario('I as an user should be able to see proper error message if search text is not in the right format', ({I,current}) => {
-  I.loginAsSystemOwner();
+  //I.loginAsSystemOwner();
```
Comment (Thomas-Geraghty) [inline src/test/functional/search-user-test.ts:21]:
```suggestion
```
Probably just worth removing this comment now.

File: src/test/functional/shared/apiHelpers.ts [hunk 2]
```diff
@@ -153,3 +153,14 @@ export const deleteUser = async (email) => {
     throw new Error(`Failed to delete user with email ${email}, http-status: ${e.response?.status}`);
   }
 };
+
+export const deleteAllTestData = async (testDataPrefix = '', userNames = [], roleNames = [], serviceNames = [], async = false) => {
+  try {
+    await axios.delete(
+      `${config.get('services.idam.url.api')}/testing-support/test-data?async=${async}&userNames=${userNames.join(',')}&roleNames=${roleNames.join(',')}&testDataPrefix=${testDataPrefix}&serviceNames=${serviceNames.join(',')}`
```
Comment (Thomas-Geraghty) [inline src/test/functional/shared/apiHelpers.ts:160]: Maybe it'd be better to build this request in a similar fashion to the other requests? using URLSearchParams object? :)

File: src/test/functional/manage-users-test.ts [hunk 3]
```diff
@@ -11,23 +10,23 @@ import {
...
-const credentials = new DataTable(['email', 'password']);
-credentials.add([testConfig.SMOKE_TEST_USER_USERNAME, testConfig.SMOKE_TEST_USER_PASSWORD]);
-credentials.add([testAccounts.superUser.email, testConfig.PASSWORD]);
-credentials.add([testAccounts.adminUser.email, testConfig.PASSWORD]);
-
-Data(credentials).Scenario('I should be able to see the active status of an user', async ({I, current}) => {
+const dashboardUserEMAIL = testConfig.TEST_SUITE_PREFIX + randomData.getRandomEmailAddress();
+BeforeSuite(async () => {
+  const dashboardUser = await createUserWithRoles(dashboardUserEMAIL, testConfig.PASSWORD, testConfig.USER_FIRSTNAME, []);
+  console.log(dashboardUser.id, dashboardUser.email);
```
Comment (Thomas-Geraghty) [inline src/test/functional/manage-users-test.ts:21]:
```suggestion
```
I'm presuming this is just leftover from dev? :)

File: src/test/functional/shared/apiHelpers.ts (hunk 2 above)
Comment (Thomas-Geraghty) [reply src/test/functional/shared/apiHelpers.ts:160]: Actually we can leave this for another day

**Comment by Thomas-Geraghty** [comment]
https://github.com/hmcts/idam-user-dashboard/blob/5a8ab461d89547899bbd05a02a45f580f7974380/src/test/functional/shared/random-data.ts#L18

In the random test data, can we change this email domain to `@local.test` - otherwise the gov.uk servers will probably try handle this data

File: src/test/functional/search-user-test.ts (hunk 1 above)
Comment (ashwini-mv) [reply src/test/functional/search-user-test.ts:21]: Removed

File: src/test/functional/manage-users-test.ts (hunk 3 above)
Comment (ashwini-mv) [reply src/test/functional/manage-users-test.ts:21]: Yes, fixed

**Review by Thomas-Geraghty — APPROVED** [review]

**Pull request merged**